### PR TITLE
Remove `typedesc` class for records

### LINKED ISSUE #41851: [Task]: Remove `typedesc` class create for records
### Description

In the earlier implementation includes a `typedesc` class for storing default values of record fields. This was initially necessary to retain default values within the `typedesc`. However, with the recent work to creating closures for record defaults, the `typedesc` class is no longer required and can be safely removed.

### Describe your task(s)

_No response_

### Related area

-> Runtime

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by chiranSachintha**
Fixes #41851

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by MaryamZi** [comment]
@chiranSachintha, this will require changes since we decided to have a record typedesc class, right? Shall we close this and reopen once it is ready to be reviewed?

**Comment by chiranSachintha** [comment]
> @chiranSachintha, this will require changes since we decided to have a record typedesc class, right? Shall we close this and reopen once it is ready to be reviewed?

What we discussed is to introducing default values to map within the type description (typedesc), aligning with the approach used in the previous implementation, and remove the separate classes generated for the typedesc of records. Therefore, this fix is required to implement those changes.

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/TypedescValueImpl.java [hunk 1]
```diff
@@ -97,8 +97,8 @@ public Object instantiate(Strand s) {
     @Override
     public Object instantiate(Strand s, BInitialValueEntry[] initialValues) {
         Type referredType = getImpliedType(this.describingType);
-        if (referredType.getTag() == TypeTags.MAP_TAG) {
-            return new MapValueImpl(this.describingType, (BMapInitialValueEntry[]) initialValues);
+        if (referredType.getTag() == TypeTags.MAP_TAG || referredType.getTag() == TypeTags.RECORD_TYPE_TAG) {
+            return new MapValueImpl(this.describingType, (BMapInitialValueEntry[]) initialValues, this);
         } else if (referredType.getTag() == TypeTags.TUPLE_TAG) {
```
Comment (MaryamZi) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/TypedescValueImpl.java:102]: Extract `referredType.getTag()` out to a variable.

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/error/ErrorTest.java [hunk 2]
```diff
@@ -84,8 +84,8 @@ public void testIndirectErrorCtor() {
         Object returns = BRunUtil.invoke(errorTestResult, "testIndirectErrorConstructor");
         BArray errors = (BArray) returns;
         Assert.assertEquals(errors.size(), 4);
-        Assert.assertEquals(errors.get(0).toString(), "error UserDefErrorTwoA (\"arg\",message=\"\",data={})");
-        Assert.assertEquals(errors.get(1).toString(), "error UserDefErrorTwoA (\"arg\",message=\"\",data={})");
+        Assert.assertEquals(errors.get(0).toString(), "error UserDefErrorTwoA (\"arg\",data={},message=\"\")");
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/error/ErrorTest.java:87]: Can you please mention (as a reply here) what kind of order changes we get with this change?

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/TypedescValueImpl.java [hunk 3]
```diff
@@ -97,8 +97,8 @@ public Object instantiate(Strand s) {
     @Override
     public Object instantiate(Strand s, BInitialValueEntry[] initialValues) {
         Type referredType = getImpliedType(this.describingType);
```
Comment (dulajdilshan) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/TypedescValueImpl.java:99]: You can directly get the `tag` and use it

```java
int tag = getImpliedType(this.describingType).;
```

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/error/ErrorTest.java (hunk 2 above)
Comment (MaryamZi) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/error/ErrorTest.java:87]: Ping on this. Can we check if we can avoid the order change?

**Comment by rdulmina** [comment]
Closing this PR since if we remove the typedesc class we need to find a way to preserve the value class for the corresponding typedesc. Hence we decided to keep the class as it is

**Pull request closed without merging**